tests/cypress/tests/unit/query_syntactic_tree_test.ts

```diff
@@ -1,38 +1,38 @@
-            expect(new SyntaxTree(`'myField\\' eq num(0)`).isValid).to.equal(false);
-            expect(new SyntaxTree(`"myField\\" eq num(0)`).isValid).to.equal(false);
-            expect(new SyntaxTree('`myField\\` eq num(0)').isValid).to.equal(false);
-            expect(new SyntaxTree(`\\'myField' eq num(0)`).isValid).to.equal(false);
-            expect(new SyntaxTree(`\\"myField" eq num(0)`).isValid).to.equal(false);
-            expect(new SyntaxTree('\\`myField` eq num(0)').isValid).to.equal(false);
+            expect(new QuerySyntaxTree(`{myField} eq 0`).isValid).to.equal(true);
+            expect(new QuerySyntaxTree(`{'myField'} eq 0`).isValid).to.equal(true);
+            expect(new QuerySyntaxTree(`{"myField"} eq 0`).isValid).to.equal(true);
+            expect(new QuerySyntaxTree('{`myField`} eq 0').isValid).to.equal(true);
+            expect(new QuerySyntaxTree('{\\{myField\\}} eq 0').isValid).to.equal(true);
+            expect(new QuerySyntaxTree('{{myField}} eq 0').isValid).to.equal(false);
+            expect(new QuerySyntaxTree(`{myField eq 0`).isValid).to.equal(false);
+            expect(new QuerySyntaxTree(`myField} eq 0`).isValid).to.equal(false);
+            expect(new QuerySyntaxTree(`myField eq 0`).isValid).to.equal(false);
         });
 
         it('does not support badly formed expression', () => {
-            expect(new SyntaxTree(`myField eq 'value'`).isValid).to.equal(true);
-            expect(new SyntaxTree(`myField eq "value"`).isValid).to.equal(true);
-            expect(new SyntaxTree('myField eq `value`').isValid).to.equal(true);
-            expect(new SyntaxTree(`myField eq 'value\\'`).isValid).to.equal(false);
-            expect(new SyntaxTree(`myField eq "value\\"`).isValid).to.equal(false);
-            expect(new SyntaxTree('myField eq `value\\`').isValid).to.equal(false);
-            expect(new SyntaxTree(`myField eq \\'value'`).isValid).to.equal(false);
-            expect(new SyntaxTree(`myField eq \\"value"`).isValid).to.equal(false);
-            expect(new SyntaxTree('myField eq \\`value`').isValid).to.equal(false);
+            expect(new QuerySyntaxTree(`{myField} eq 'value'`).isValid).to.equal(true);
+            expect(new QuerySyntaxTree(`{myField} eq "value"`).isValid).to.equal(true);
+            expect(new QuerySyntaxTree('{myField} eq `value`').isValid).to.equal(true);
+            expect(new QuerySyntaxTree(`{myField} eq 'value\\'`).isValid).to.equal(true);
+            expect(new QuerySyntaxTree(`{myField} eq 'value\\''`).isValid).to.equal(true);
+            expect(new QuerySyntaxTree(`{myField} eq "value\\"`).isValid).to.equal(true);
+            expect(new QuerySyntaxTree('{myField} eq `value\\`').isValid).to.equal(true);
+            expect(new QuerySyntaxTree(`{myField} eq \\'value'`).isValid).to.equal(false);
+            expect(new QuerySyntaxTree(`{myField} eq \\"value"`).isValid).to.equal(false);
```

Comment:
non-blocking style point - I find this kind of test easier to read (and more likely that more thorough cases will be added) if they're written as a loop (one for all the passes, another for all the fails):
```js
[
  `{myField} eq 'value'`,
  `{myField} eq "value"`,
  '{myField} eq `value`',
  `{myField} eq 'value\\'`,
  `{myField} eq 'value\\''`,
  `{myField} eq "value\\"`,
  '{myField} eq `value\\`'
].forEach(v => {
  expect(new QuerySyntaxTree(v).isValid).to.equal(true, v);
});
```
(assuming the extra msg arg `.equal(true, v)` works so you know which one failed)